### DISCUSSION TITLE
Hand and controller rendering and interaction broken after upgrading to latest version of MRTK

### DISCUSSION THREAD

**Discussion by halfpasttense** [MixedRealityToolkit/MixedRealityToolkit-Unity#660] [Q&A]
Just upgraded my Unity project to the latest version:

![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/40369329/7717649c-a507-416d-b55d-886aad86b83e)

Project runs and compiles but when running natively on Quest Pro neither the controllers nor hands render, nor do any of the interaction pointers. When I run it through Quest Link from the Unity Editor the controllers do not render, the left hand does render but not the right hand, and some of the interaction pointers do render but don't work right. These are all the errors I see in the logs:

```
ArgumentNullException: Value cannot be null.
Parameter name: type
System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic, System.Boolean wrapExceptions) (at <1f66344f2f89470293d8b67d71308c07>:0)
System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic) (at <1f66344f2f89470293d8b67d71308c07>:0)
System.Activator.CreateInstance (System.Type type) (at <1f66344f2f89470293d8b67d71308c07>:0)
MixedReality.Toolkit.SpatialManipulation.ObjectManipulator.Awake () (at Library/PackageCache/org.mixedrealitytoolkit.spatialmanipulation@9d90b09a2a11/ObjectManipulator/ObjectManipulator.cs:626)

Could not find active control after binding resolution
UnityEngine.InputSystem.LowLevel.NativeInputRuntime/<>c__DisplayClass7_0:<set_onUpdate>b__0 (UnityEngineInternal.Input.NativeInputUpdateType,UnityEngineInternal.Input.NativeInputEventBuffer*)
UnityEngineInternal.Input.NativeInputSystem:NotifyUpdate (UnityEngineInternal.Input.NativeInputUpdateType,intptr)

ArgumentNullException: Value cannot be null.
Parameter name: type
System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic, System.Boolean wrapExceptions) (at <1f66344f2f89470293d8b67d71308c07>:0)
System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic) (at <1f66344f2f89470293d8b67d71308c07>:0)
System.Activator.CreateInstance (System.Type type) (at <1f66344f2f89470293d8b67d71308c07>:0)
MixedReality.Toolkit.SpatialManipulation.ObjectManipulator.Awake () (at Library/PackageCache/org.mixedrealitytoolkit.spatialmanipulation@9d90b09a2a11/ObjectManipulator/ObjectManipulator.cs:626)
UnityEngine.Events.UnityEvent:Invoke()
MixedReality.Toolkit.SpatialManipulation.HandConstraint:SolverUpdate() (at Library/PackageCache/org.mixedrealitytoolkit.spatialmanipulation@9d90b09a2a11/Solvers/HandConstraint.cs:264)
MixedReality.Toolkit.SpatialManipulation.Solver:SolverUpdateEntry() (at Library/PackageCache/org.mixedrealitytoolkit.spatialmanipulation@9d90b09a2a11/Solvers/Solver.cs:290)
MixedReality.Toolkit.SpatialManipulation.SolverHandler:LateUpdate() (at Library/PackageCache/org.mixedrealitytoolkit.spatialmanipulation@9d90b09a2a11/Solvers/SolverHandler.cs:369)
```

**Comment by AMollis** [comment]
@keveleigh , I believe you updated something in this area with the last release?

**Reply by keveleigh** [reply]
I didn't make any changes to `ObjectManipulator`. Looks like the line that's failing hasn't been changed in a while:

![image](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/3580640/c6495328-d78f-49d9-b6ec-a264a97e4de8)

Looks like the serialized ITransformSmoothingLogic went null somehow? Did we remove or rename a type in this release or something?

> Could not find active control after binding resolution
> UnityEngine.InputSystem.LowLevel.NativeInputRuntime/<>c__DisplayClass7_0:<set_onUpdate>b__0 (UnityEngineInternal.Input.NativeInputUpdateType,UnityEngineInternal.Input.NativeInputEventBuffer*)
UnityEngineInternal.Input.NativeInputSystem:NotifyUpdate (UnityEngineInternal.Input.NativeInputUpdateType,intptr)

This one seems strange too. It's _possible_ it's a knock-on effect, but I'm not sure exactly how.
I wonder if the Unity OpenXR package was indirectly updating and something needs changing with the interaction profiles in Unity's OpenXR settings. I know they were migrating from one Quest profile to another to update the class name, but I doubt they'd intentionally break something like this.
